src/BuildScriptGenerator/Python/PythonPlatform.cs

```diff
@@ -243,7 +243,16 @@ public PlatformDetectorResult Detect(RepositoryContext context)
                     = compressedVirtualEnvFileName;
             }
 
-            this.TryLogDependencies(pythonVersion, context.SourceRepo);
+            var customRequirementsTxtPath = this.pythonScriptGeneratorOptions.CustomRequirementsTxtPath;
+            var requirementsTxtPath = customRequirementsTxtPath == null ? PythonConstants.RequirementsFileName : customRequirementsTxtPath;
+            if (!context.SourceRepo.FileExists(requirementsTxtPath))
```

Comment:
Could you please confirm that having a requirements.txt file is actually required for all Python applications?

Reply:
The code we had previously wasn't throwing an exception, and was just exiting TryLogDependencies instead

Reply:
Ah good catch -- let me clean that up.